### PR TITLE
panel pass kwargs to the widget

### PR DESCRIPTION
**Pull request by henriqueribeiro**
Implements https://github.com/pyviz/panel/issues/456

Following @philippjfr suggestion, now it's also possible to pass a dictionary to the widgets. This dictionary will be passed as kwargs to the widget itself.

### CODE REVIEW

**Comment by philippjfr** [comment]
@jbednar Do we prefer 'widget' or 'type' to override the widget type as a dictionary key?

**Comment by philippjfr** [comment]
We should document this in the widget override section of the Param user guide.

**Comment by philippjfr** [comment]
Thanks for the PR @henriqueribeiro, hugely appreciated!

**Comment by jbednar** [comment]
Isn't `type` a reserved word?  I do prefer `type` to `widget`, I think, but I'd have to see an example of usage (as you requested).

**Comment by philippjfr** [comment]
Here's an example:

```
class Example(param.Parameterized):

    test = param.ListSelector(default=['a'], objects=['a', 'b', 'c', 'd', 'e', 'f'], height=10)

    @param.depends("test")
    def view(self):
        return self.test

example = Example()
pn.Row(pn.Param(example.param, widgets={'test': {'type': pn.widgets.CheckBoxGroup, 'inline': False}} , example.view)
```

**Comment by jbednar** [comment]
I have no preference between `type` and `widget` in this case; they are both accurate.  We should be sure to clarify in the same section of the docs that it's a lot easier to change the `inline` parameter at the class level, for the cases where that is sufficient:

```
import panel.widgets as pw
pw.CheckBoxGroup.inline=False
```

**Comment by henriqueribeiro** [comment]
Hey! I do agree with 'type' instead of widget. I will change it and will document it. I just wanted you to check if you agree with this first version :smile:

File: panel/param.py [hunk 1]
```diff
@@ -285,9 +285,17 @@ def update_pane(change, parameter=pname):
...
         """Get widget for param_name"""
         p_obj = self.object.param[p_name]
+        kw_widget = {}
 
         if self.widgets is None or p_name not in self.widgets:
             widget_class = self.widget_type(p_obj)
+        elif isinstance(self.widgets[p_name], dict):
+            if 'widget' in self.widgets[p_name]:
+                widget_class = self.widgets[p_name]['widget']
+                self.widgets[p_name].pop('widget')
```
Comment (philippjfr) [inline panel/param.py]: You can do this in a single line:

```python
widget_class = self.widgets[p_name].pop('widget')
```

**Comment by philippjfr** [comment]
Looks good to me. Would love to see some documentation and a test but won't let that hold up merging.

**Comment by henriqueribeiro** [comment]
I will add some tests later

**Comment by philippjfr** [comment]
Thanks so much!

**Pull request merged**